src/core/components/push_payload.js

```diff
@@ -0,0 +1,537 @@
+/* @flow */
+import { APNS2Configuration, APNS2Target } from '../flow_interfaces';
+
+class BaseNotificationPayload {
+  _subtitle: ?string;
+  _payload: Object;
+  _badge: ?number;
+  _sound: ?string;
+  _title: ?string;
+  _body: ?string;
+
+  get payload() {
+    return this._payload;
+  }
+
+  set title(value: ?string) {
```

Comment:
Do we ever want to set a `null` value vs an `undefined` value?

This parameter setting will allow both.

Reply:
This annotation allow user to remove value from payload, if he want to.

Reply:
This specifically allows setting a `string` , `null` or `undefined` . So i'm asking if the values for `null` AND `undefined` are allowed values.

If we only want `string` or `undefined` then we should use: `set title(value?: string) {` instead.

Reply:
@davidnub user can pass `null` or `undefined` to reset value, so it won't be included into payload. Both `null` and `undefined` can be used in _if_ condition to check whether value has been set or not. 
We can limit it to `undefined` only, but will looks strange in support: _if you want to clean up this field, pass `undefined` to this property_. `null` in this case will be better, because it is assignment value which say: no value.

Reply:
I don't think using `string | null` is necessary, i'm most asking if we set `null` or `undefined` for any specific reason. It sounds like the reason you're giving me is that we're using it for "truthy" checks. Wherein we allow `null` and `undefined` to be used as *falsey* values.

So that means somewhere else we check for the *falsey* values and remove the properties is that correct? We won't ever send `null` or `undefined` to the API? <-- This is my question.

Reply:
@davidnub looks like we end up with miss understanding each other. Looks like this statement was only clear to me: “ user can pass null or undefined to reset value, so it won't be included into payload.”.
Yes, if user pass `null` (which should be used for assignment, when `undefined` is default state of unassigned value), then field value will be excluded from payload (user message won’t include it)

Reply:
Ok then!